pkg/btf/validation.go

```diff
@@ -173,72 +173,90 @@ func getKernelType(arg btf.Type) string {
 	return arg.TypeName()
 }
 
-func typesCompatible(specTy string, kernelTy string) bool {
+// checkTypesCompatibility checks if the passed types are compatible
+//
+// specTy The spec type to check
+// kernelTy The kernel type to check against
+// btfTy The BTF Type to check against instead of kernelTy
+//
+// Returns an error if the types are not compatible
+func checkTypesCompatibility(specTy string, kernelTy string, btfTy btf.Type) error {
+	if btfTy != nil {
+		kernelTy = getBtfKernelType(btfTy)
+	}
+
 	switch specTy {
 	case "size_t":
 		switch kernelTy {
 		case "size_t":
-			return true
+			return nil
 		}
 	case "char_buf", "string":
 		switch kernelTy {
 		case "const char *", "char *":
-			return true
+			return nil
 		}
 	case "char_iovec":
 		switch kernelTy {
 		case "const struct iovec *", "struct iovec *":
-			return true
+			return nil
 		}
 	case "int", "fd":
 		switch kernelTy {
 		case "unsigned int", "int", "unsigned long", "long":
-			return true
+			return nil
 		}
 	case "filename":
 		switch kernelTy {
 		case "struct filename *":
-			return true
+		case "filename":
+			return nil
 		}
 	case "file":
 		switch kernelTy {
 		case "struct file *":
-			return true
+		case "file":
+			return nil
 		}
 	case "bpf_attr":
 		switch kernelTy {
 		case "union bpf_attr *":
-			return true
+		case "bpf_attr":
+			return nil
 		}
 	case "perf_event":
 		switch kernelTy {
 		case "struct perf_event *":
-			return true
+		case "perf_event":
```

Comment:
Would it make sense to check for both `struct perf_event *` and `perf_event` here?